Framework/Geometry/test/ComponentInfoBankHelpersTest.h

```diff
@@ -175,6 +175,122 @@ class ComponentInfoBankHelpersTest : public CxxTest::TestSuite {
     TS_ASSERT_THROWS(offsetFromAncestor(compInfo, ancestorIndex, currentIndex),
                      std::invalid_argument &);
   }
+
+  void test_isAncestorOf_finds_ancestor() {
+
+    // NOTE: as defined in isAncestorOf, the root index is not included.
+
+    auto instrument = ComponentCreationHelper::createTestInstrumentRectangular2(
+        1 /*number of banks*/, 2 /*pixels*/);
+    auto wrappers = InstrumentVisitor::makeWrappers(*instrument);
+
+    // instrument cache to be used with call offsetFromAncestor
+    const auto &compInfo = (*wrappers.first);
+
+    size_t bank = 6;                                 // index of bank
+    auto pixels = compInfo.detectorsInSubtree(bank); // 4 detectors of bank
+
+    // assert the pixels are the children of the bank
+    TS_ASSERT(isAncestorOf(compInfo, bank, pixels[0]));
+    TS_ASSERT(isAncestorOf(compInfo, bank, pixels[1]));
+    TS_ASSERT(isAncestorOf(compInfo, bank, pixels[2]));
+    TS_ASSERT(isAncestorOf(compInfo, bank, pixels[3]));
+  }
+
+  void
+  test_isAncestorOf_returns_false_with_child_or_sibling_proposed_as_ancestor() {
+
+    // clang-format off
+    /*
+
+                    structure as in test:
+
+                           root
+                            |
+              source ---- sample ---- bank (siblings)
+                                       |
+                         pixel---pixel---pixel---pixel
+    */
+    // clang-format on
+
+    // NOTE: as defined in isAncestorOf, the root index is not included.
+
+    auto instrument = ComponentCreationHelper::createTestInstrumentRectangular2(
+        1 /*number of banks*/, 2 /*pixels*/);
+    auto wrappers = InstrumentVisitor::makeWrappers(*instrument);
+
+    // instrument cache to be used with call offsetFromAncestor
+    const auto &compInfo = (*wrappers.first);
+
+    size_t bank = 6;                                 // index of bank
+    auto pixels = compInfo.detectorsInSubtree(bank); // 4 detectors of bank
+    TS_ASSERT(isSaveableBank(compInfo, bank)); // assert bank is NXdetector
```

Comment:
Do not check `isSaveableBank` in this test we don't confuse the two independent functions.

Reply:
removed